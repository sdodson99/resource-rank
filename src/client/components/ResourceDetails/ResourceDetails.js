import React from 'react';
import PropTypes from 'prop-types';
import styles from './ResourceDetails.module.css';

const ResourceDetails = ({ link }) => (
  <div className={styles.ResourceDetails} data-testid="ResourceDetails">
    <div className="flex">
      <div>Link:</div>
      <div className="ml-4">
        {link && (
          <a
            className="hyperlink break-all"
            href={link}
            target="_blank"
            rel="noreferrer"
          >
            {link}
          </a>
        )}
        {!link && 'No resource link added.'}
      </div>
    </div>
  </div>
);

ResourceDetails.propTypes = {
  link: PropTypes.string,
};

ResourceDetails.defaultProps = {};

export default ResourceDetails;
