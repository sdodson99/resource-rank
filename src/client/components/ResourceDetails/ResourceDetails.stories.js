import React from 'react';
import ResourceDetails from './ResourceDetails';

export default {
  component: ResourceDetails,
  title: 'Components/ResourceDetails',
};

const Template = (args) => <ResourceDetails {...args} />;

export const Default = Template.bind({});
Default.args = {};

export const Link = Template.bind({});
Link.args = {
  link: 'https://resource-rank.web.app/',
};
