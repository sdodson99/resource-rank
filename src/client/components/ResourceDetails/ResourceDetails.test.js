import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { createRenderer } from 'react-test-renderer/shallow';
import ResourceDetails from './ResourceDetails';

describe('<ResourceDetails />', () => {
  it('should mount', () => {
    render(<ResourceDetails />);

    const resourceDetails = screen.getByTestId('ResourceDetails');

    expect(resourceDetails).toBeInTheDocument();
  });

  it('should render correctly', () => {
    const component = createRenderer().render(<ResourceDetails />);

    expect(component).toMatchSnapshot();
  });

  it('should render correctly with link', () => {
    const component = createRenderer().render(
      <ResourceDetails link="http://test.com" />
    );

    expect(component).toMatchSnapshot();
  });
});
