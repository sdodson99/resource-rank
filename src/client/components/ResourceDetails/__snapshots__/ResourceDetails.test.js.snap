// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResourceDetails /> should render correctly 1`] = `
<div
  className="ResourceDetails"
  data-testid="ResourceDetails"
>
  <div
    className="flex"
  >
    <div>
      Link:
    </div>
    <div
      className="ml-4"
    >
      No resource link added.
    </div>
  </div>
</div>
`;

exports[`<ResourceDetails /> should render correctly with link 1`] = `
<div
  className="ResourceDetails"
  data-testid="ResourceDetails"
>
  <div
    className="flex"
  >
    <div>
      Link:
    </div>
    <div
      className="ml-4"
    >
      <a
        className="hyperlink break-all"
        href="http://test.com"
        rel="noreferrer"
        target="_blank"
      >
        http://test.com
      </a>
    </div>
  </div>
</div>
`;
